funsite/utils.py

```diff
@@ -0,0 +1,33 @@
+# -*- coding: utf-8 -*-
+
+from collections import namedtuple
+
+from django.utils.translation import pgettext, ugettext as _
+
+BreadcrumbsItem = namedtuple('BreadcrumbsItem', ['path', 'name'])
+
+
+def breadcrumbs(url, current_page):
+    """
+    Return a list of pair (path, i18ned name) of the breadcrumb for current url
+    The tricky part is the replacement of edX's courses page by our.
+    See tests.test_breadcrumbs to understand awaited behaviour
+    """
+
+    mapping = {
+        'contact': _("Contact"),
+        'courses': _("All courses"),    # edX page
+        'cours': _("All courses"),      # FUN page
+        'news': pgettext("fun-news", "News"),
+        'universities': _("Universities"),
+    }
+    result = [BreadcrumbsItem(path='/', name=_(u"Home"))]
+
+    items = url.split('/')[1:]  # split and remove first backslash
+    if len(items) > 1 and items[1]:
+        if items[0] == 'courses':
+            items[0] = 'cours'
+        result.append(BreadcrumbsItem(path='/%s/' % items[0], name=mapping[items[0]]))
```

Comment:
En fait, cette fonction suppose que `url` est de la forme `path/id` ou `path` est une url valide. Si ce n'est pas le cas, l'appel à la fonction va résulter en une url qui donne une 404.
On pourrait ajouter une condition `if items[0] in matching:...` avant de faire `result.append(...)` ?